shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -477,6 +488,72 @@ public boolean performEditorAction(int actionCode) {
     return true;
   }
 
+  @Override
+  @TargetApi(25)
+  @RequiresApi(25)
+  public boolean commitContent(InputContentInfo inputContentInfo, int flags, Bundle opts) {
+    // Ensure permission is granted.
+    if (Build.VERSION.SDK_INT >= 25
+        && (flags & InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0) {
+      try {
+        inputContentInfo.requestPermission();
+      } catch (Exception e) {
+        return false;
+      }
+    } else {
+      return false;
```

Comment:
What will this end up looking like in the framework if this isn't supported by the device for some reason?  Ideally an app developer would be able to gracefully handle this situation.

Reply:
There's a toast message displayed that "<app> doesn't support image insertion here". It will display this toast any time this function returns false. To my knowledge, it cannot be customized and is an Android framework thing.

Reply:
Ok sounds like that's the same as native so that should be fine 👍